packages/connect/src/protocol-connect/handler-factory.spec.ts

```diff
@@ -12,28 +12,30 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-import {
-  Int32Value,
-  Message,
-  MethodKind,
-  StringValue,
-} from "@bufbuild/protobuf";
-import type { MethodInfo, ServiceType } from "@bufbuild/protobuf";
+import type { MethodInfo } from "@bufbuild/protobuf";
+import { Int32Value, MethodKind, StringValue } from "@bufbuild/protobuf";
 import { createHandlerFactory } from "./handler-factory.js";
+import type { MethodImpl } from "../implementation.js";
 import { createMethodImplSpec } from "../implementation.js";
-import type { HandlerContext, MethodImpl } from "../implementation.js";
 import type { UniversalHandlerOptions } from "../protocol/index.js";
-import { errorFromJsonBytes } from "./error-json.js";
+import {
+  createAsyncIterable,
+  createUniversalHandlerClient,
+  pipeTo,
+  sinkAll,
+} from "../protocol/index.js";
 import { ConnectError } from "../connect-error.js";
-import { Code } from "../code.js";
-import { endStreamFromJson } from "./end-stream.js";
 import {
   createAsyncIterableBytes,
   readAllBytes,
 } from "../protocol/async-iterable-helper.spec.js";
+import { Code } from "../code.js";
+import { errorFromJsonBytes } from "./error-json.js";
+import { endStreamFromJson } from "./end-stream.js";
+import { createTransport } from "./transport.js";
 
 describe("createHandlerFactory()", function () {
-  const testService: ServiceType = {
+  const testService = {
     typeName: "TestService",
     methods: {
```

Comment:
thought: would this test suite be an appropriate place to also have a bidi test (right now it's just unary and server-streaming)?

Reply:
This would become a very hard to maintain x-thousand LoC test file if it tried to get decent coverage, and we would still need coverage for running on an actual network with Node.js etc. See comment above about crosstests. They don't cover this case for bidi yet though.